Neos.ContentRepository.Core/Classes/Infrastructure/Property/PropertyConverter.php

```diff
@@ -72,26 +67,35 @@ private function serializePropertyValue(
             $nodeTypeName
         );
 
-        if ($propertyValue !== null) {
-            try {
-                $propertyValue = $this->serializer->normalize($propertyValue);
-            } catch (NotEncodableValueException | NotNormalizableValueException $e) {
-                throw new \RuntimeException(
-                    'TODO: There was a problem serializing ' . get_class($propertyValue),
-                    1594842314,
-                    $e
-                );
-            }
-
-            return new SerializedPropertyValue(
-                $propertyValue,
-                $propertyType->value
+        if ($propertyValue === null) {
+            // should not happen, as we must separate regular properties and unsets beforehand!
+            throw new \RuntimeException(
+                sprintf('Property %s with value "null" cannot be serialized as unsets are treated differently.', $propertyName->value),
+                1707578784
             );
-        } else {
-            // $propertyValue == null and defined in node types (we have a resolved type)
-            // -> we want to set the $propertyName to NULL
-            return new SerializedPropertyValue(null, $propertyType->value);
         }
+
+        try {
+            $serializedPropertyValue = $this->serializer->normalize($propertyValue);
+        } catch (NotEncodableValueException | NotNormalizableValueException $e) {
+            throw new \RuntimeException(
+                sprintf('There was a problem serializing property %s with value "%s".', $propertyName->value, get_debug_type($propertyValue)),
+                1594842314,
+                $e
+            );
+        }
+
+        if ($serializedPropertyValue === null) {
```

Comment:
Mmm, I find this tricky because I could see edge cases were serialization might result in a NULL value without the user really being able to control that?

Reply:
Yes i know. I briefly discussed this with bastian freely after lets see :D

The thing is `node->hasProperty` will become more hard to describe and understand if `node->getProperty` can return `null` values if the node _has_ the property. What do you say?

But in the end we can always loosen this ...

Reply:
I think the tricky part is that getProperty wil need to be able to return null, think policies, lets say you have an entity that not everyone can see, this would be reflected in the property, and the unserialiser shouldn't be able to unserialize it for some viewers.

Reply:
I think we should open up the discussion when this case happens. Being more restrictive here in the first place seems valid?
Having an serializer enforce policies is also weird ....

Reply:
Actually I’m sure we need to catch this here 100%

if a serializer failed at write/command time that is okay

But the code below in deserializePropertyValue might apply to your concerns where I still find it confusing … is possible in the existing system for example to entity privilege protect an image or any asset? If so we probably do need to talk about that …

Reply:
Yeah okay, reverted the read side error throwing as this is not handleable by the user. The write side is fine imo and desired.

Discussion how this might affect our api and confuse users can be continued here: https://github.com/neos/neos-development-collection/issues/4911